In [ ]:
import numpy
numpy.set_printoptions(threshold=numpy.inf)

In [ ]:
from google.colab import drive 
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
with open('/content/drive/MyDrive/exp5-3_english/test.text.txt', 'r') as f:
    train_text = f.read().split("\n")

#train_textの最後に''があるため削除
train_text = train_text[:-1]
print(train_text)

['A girl is styling her hair.\tA girl is brushing her hair.', 'A group of men play soccer on the beach.\tA group of boys are playing soccer on the beach.', "One woman is measuring another woman's ankle.\tA woman measures another woman's ankle.", 'A man is cutting up a cucumber.\tA man is slicing a cucumber.', 'A man is playing a harp.\tA man is playing a keyboard.', 'A woman is cutting onions.\tA woman is cutting tofu.', 'A man is riding an electric bicycle.\tA man is riding a bicycle.', 'A man is playing the drums.\tA man is playing the guitar.', 'A man is playing guitar.\tA lady is playing the guitar.', 'A man is playing a guitar.\tA man is playing a trumpet.', 'A man is playing a guitar.\tA man is playing a trumpet.', 'A man is cutting an onion.\tA man cuts an onion.', 'A man is cycling.\tA man is talking.', 'A man is slicing open a fish.\tA man is cutting up a fish.', 'A man is slicing a tomato.\tA man is slicing a bun.', 'A man is playing a guitar.\tA man is playing a keyboard.', 

In [ ]:
# インストール　英語の定番の単語分割器
! pip install nltk

import nltk
nltk.download("punkt") # 分かち書き
nltk.download("averaged_perceptron_tagger") # 品詞の取得
#https://qiita.com/m__k/items/ffd3b7774f2fde1083fa

nltk.download('wordnet') # エラーに記載されていたものを追加
nltk.download('omw-1.4') # エラーに記載されていたものを追加
#https://www.haya-programming.com/entry/2018/03/25/203836  語幹化  dogs -> dog

from nltk import word_tokenize
from nltk import pos_tag

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


[nltk_data] Downloading package punkt to /root/nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package averaged_perceptron_tagger to
[nltk_data]     /root/nltk_data...
[nltk_data]   Package averaged_perceptron_tagger is already up-to-
[nltk_data]       date!
[nltk_data] Downloading package wordnet to /root/nltk_data...
[nltk_data]   Package wordnet is already up-to-date!
[nltk_data] Downloading package omw-1.4 to /root/nltk_data...


In [ ]:
#ストップワード
from nltk.corpus import stopwords

nltk.download('stopwords')
stop_words = stopwords.words('english')
print(stop_words)

stop_words = stop_words[:85]
stop_words.append('on')
stop_words.append('in')
stop_words.append("'s")
print(stop_words)
# https://nashidos.hatenablog.com/entry/2020/08/12/205119

['i', 'me', 'my', 'myself', 'we', 'our', 'ours', 'ourselves', 'you', "you're", "you've", "you'll", "you'd", 'your', 'yours', 'yourself', 'yourselves', 'he', 'him', 'his', 'himself', 'she', "she's", 'her', 'hers', 'herself', 'it', "it's", 'its', 'itself', 'they', 'them', 'their', 'theirs', 'themselves', 'what', 'which', 'who', 'whom', 'this', 'that', "that'll", 'these', 'those', 'am', 'is', 'are', 'was', 'were', 'be', 'been', 'being', 'have', 'has', 'had', 'having', 'do', 'does', 'did', 'doing', 'a', 'an', 'the', 'and', 'but', 'if', 'or', 'because', 'as', 'until', 'while', 'of', 'at', 'by', 'for', 'with', 'about', 'against', 'between', 'into', 'through', 'during', 'before', 'after', 'above', 'below', 'to', 'from', 'up', 'down', 'in', 'out', 'on', 'off', 'over', 'under', 'again', 'further', 'then', 'once', 'here', 'there', 'when', 'where', 'why', 'how', 'all', 'any', 'both', 'each', 'few', 'more', 'most', 'other', 'some', 'such', 'no', 'nor', 'not', 'only', 'own', 'same', 'so', 'than', '

[nltk_data] Downloading package stopwords to /root/nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


In [ ]:
#前処理 単語分割から呼び出す　共通の処理
def prepro(word):
    if word.isalpha() and not word.islower(): #すべて小文字に->辞書の都合
        word = word.lower()
    if word.isnumeric():
        word = '0'
    if word == "n't":
        word = 'not'
    if word == "'ll":
        word = 'will'
    if word == "'ve":
        word = 'have'
    return word

In [ ]:
# NLTK 単語分割を行う関数を定義
from nltk.stem import WordNetLemmatizer
lemmatizer = WordNetLemmatizer()

def nltk_split(text):
    spword = list()
    for token in pos_tag(word_tokenize(text)):
        word, pos = token
        word = prepro(word)
        word = lemmatizer.lemmatize(word, pos="v") # https://qiita.com/m__k/items/ffd3b7774f2fde1083fa
        if word != ',' and word != ':' and word != '.' and word != '?':
            if not word in stop_words:
                    spword.append(word)
    return spword

In [ ]:
# Moses Tokenizer インストール
! pip install mosestokenizer

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
  Created wheel for mosestokenizer: filename=mosestokenizer-1.2.1-py3-none-any.whl size=49210 sha256=571cb15b07bafb12fd7735b87196f70f285da7708bd5d6dbd7ed0996fca93096
  Stored in directory: /root/.cache/pip/wheels/b0/35/f7/af1258779a0b890abc3c79481460c597cb1f3659d0603cfb9d
  Created wheel for docopt: filename=docopt-0.6.2-py2.py3-none-any.whl size=13723 sha256=4dca8813bc0ce276c3b7f8ffe88b6d11a17aab00d89d6e695ea1565666a847d9
  Stored in directory: /root/.cache/pip/wheels/72/b0/3f/1d95f96ff986c7dfffe46ce2be4062f38ebd04b506c77c81b9
  Created wheel for toolwrapper: filename=toolwrapper-2.1.0-py3-none-any.whl size=3378 sha256=293b5314929af0442a3f782c6f5ccec80e28bde81a2c4855725b8eacf4362e8f
  Stored in directory: /root/.cache/pip/wheels/c5/4f/33/54741ffe08e38ececb1d28068a153729b4fe820bafa0a0691f
  Created wheel for uctools: filename=uctools-1.3.0-py3-none-any.whl size=6185 sha256=09e0a724b04ef7d

In [ ]:
from mosestokenizer import MosesTokenizer

# Moses Tokenizer 単語分割を行う関数を定義
def mt_split(text):
    tokenizer = MosesTokenizer("en", "no-escape")
    spword = list()
    for word in tokenizer(text):
        word = prepro(word)
        if word != ',' and word != ':' and word != '.' and word != '?':
            if not word in stop_words:
                    spword.append(word)
    return spword

In [ ]:
#Stanza インストール
! pip install stanza

import stanza
english_tokenizer = stanza.Pipeline("en", processors="tokenize, pos")

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 691 kB 5.0 MB/s 
     |████████████████████████████████| 240 kB 50.3 MB/s 
  Created wheel for emoji: filename=emoji-2.2.0-py3-none-any.whl size=234927 sha256=9320601991e3f3f9b014fc24738c4ab9d2c9d7ebf69debe375c19fb86130dd38
  Stored in directory: /root/.cache/pip/wheels/f3/e3/f2/1de1c2e3ed742e1df73e0f15d58864e50c7e64f607b548d6cf
Successfully built emoji


INFO:stanza:Checking for updates to resources.json in case models have been updated.  Note: this behavior can be turned off with download_method=None or download_method=DownloadMethod.REUSE_RESOURCES


INFO:stanza:Loading these models for language: en (English):
| Processor | Package  |
------------------------
| tokenize  | combined |
| pos       | combined |

INFO:stanza:Use device: cpu
INFO:stanza:Loading: tokenize
INFO:stanza:Loading: pos
INFO:stanza:Done loading processors!


In [ ]:
# Stanza 単語分割を行う関数を定義
def st_split(text):
    spword = list()
    for sentence in english_tokenizer(text).sentences:
        for token in sentence.words:
            word = token.text
            #pos = token.xpos
            word = prepro(word)
            if word != ',' and word != ':' and word != '.' and word != '?':
                if not word in stop_words:
                        spword.append(word)
    return spword

In [ ]:
text_nltk1 = list() # 左の文を追加してくための空リスト
text_nltk2 = list() # 右の文を追加してくための空リスト
text_mt1 = list() 
text_mt2 = list()
text_st1 = list() 
text_st2 = list()

for t_text in train_text:
    txt1, txt2 = t_text.split("\t")
    #text1とtext2が類似度におけるペア
    #これを文章から単語に分割していく

    text_nltk1.append(nltk_split(txt1)) # text1について 単語分割 NLTK
    text_nltk2.append(nltk_split(txt2)) # text2について 単語分割 NLTK

    text_mt1.append(mt_split(txt1)) # text1について 単語分割 Moses Tokenizer
    text_mt2.append(mt_split(txt2)) # text2について 単語分割 Moses Tokenizer

    text_st1.append(st_split(txt1)) # text1について 単語分割 Stanza
    text_st2.append(st_split(txt2)) # text2について 単語分割 Stanza

In [ ]:
#BoW

# 語彙（扱いたい全ての単語）
vocab = set()
for text1_ in text_nltk1: # NLTKで分割したもの
    for word in text1_:
        vocab.add(word)

for text2_ in text_nltk2: # NLTKで分割したもの
    for word in text2_:
        vocab.add(word)

# 各単語にIDを割り当てる 日課：0番目, ...
word2id = dict()
for word in vocab:
    word2id[word] = len(word2id)

def vectorize_bow(words, vocab, word2id):
    vector = np.zeros(len(vocab))
    for word in words:
        word = prepro(word)
        if word in vocab:
            vector[word2id[word]] = 1
    return vector

In [ ]:
tokenizer = MosesTokenizer("en", "no-escape")
# 辞書(回数)作成のための わかち
def wakati(text):
    after = list()
    for word in tokenizer(text):
        after.append(word)
    return after   

In [ ]:
import math
from collections import defaultdict

freq_dict = defaultdict(int) #頻度
prob_dict = defaultdict(float) #確率
with open('/content/drive/MyDrive/exp5-3_english/en_wiki.sent.10m.txt', 'r') as f:
    for line in f:
        wakatitext = wakati(line)
        for word in wakatitext:
            prob_dict[word] += 1
        for word in set(wakatitext):
            freq_dict[word] += 1

In [ ]:
#頻度の計算
total = sum(prob_dict.values())
for word, freq in prob_dict.items(): #辞書からkeyとvalueを取り出し
    prob_dict[word] = freq / total #確率で置き換え

#idfの計算
def idf(word):
    if freq_dict[word] == 0:
        return 1
    else:
        return 1 + math.log( 10000000 / freq_dict[word])

In [ ]:
print(text_nltk1)

[['girl', 'styling', 'hair'], ['group', 'men', 'play', 'soccer', 'beach'], ['one', 'woman', 'measuring', 'another', 'woman', 'ankle'], ['man', 'cutting', 'up', 'cucumber'], ['man', 'playing', 'harp'], ['woman', 'cutting', 'onions'], ['man', 'riding', 'electric', 'bicycle'], ['man', 'playing', 'drums'], ['man', 'playing', 'guitar'], ['man', 'playing', 'guitar'], ['man', 'playing', 'guitar'], ['man', 'cutting', 'onion'], ['man', 'cycling'], ['man', 'slicing', 'open', 'fish'], ['man', 'slicing', 'tomato'], ['man', 'playing', 'guitar'], ['baby', 'panda', 'goes', 'down', 'slide'], ['man', 'singing', 'playing', 'guitar'], ['man', 'attacks', 'woman'], ['man', 'driving', 'car'], ['woman', 'cutting', 'tofu'], ['woman', 'styling', 'hair'], ['two', 'zebras', 'play', 'open', 'field'], ['man', 'cutting', 'potato'], ['man', 'slicing', 'onion'], ['man', 'dancing'], ['man', 'riding', 'motorcycle'], ['woman', 'slicing', 'garlics'], ['man', 'speaking'], ['little', 'boy', 'singing', 'playing', 'guitar'],

In [ ]:
print(text_mt1)

[['girl', 'styling', 'hair'], ['group', 'men', 'play', 'soccer', 'beach'], ['one', 'woman', 'measuring', 'another', 'woman', 'ankle'], ['man', 'cutting', 'up', 'cucumber'], ['man', 'playing', 'harp'], ['woman', 'cutting', 'onions'], ['man', 'riding', 'electric', 'bicycle'], ['man', 'playing', 'drums'], ['man', 'playing', 'guitar'], ['man', 'playing', 'guitar'], ['man', 'playing', 'guitar'], ['man', 'cutting', 'onion'], ['man', 'cycling'], ['man', 'slicing', 'open', 'fish'], ['man', 'slicing', 'tomato'], ['man', 'playing', 'guitar'], ['baby', 'panda', 'goes', 'down', 'slide'], ['man', 'singing', 'playing', 'guitar'], ['man', 'attacks', 'woman'], ['man', 'driving', 'car'], ['woman', 'cutting', 'tofu'], ['woman', 'styling', 'hair'], ['two', 'zebras', 'play', 'open', 'field'], ['man', 'cutting', 'potato'], ['man', 'slicing', 'onion'], ['man', 'dancing'], ['man', 'riding', 'motorcycle'], ['woman', 'slicing', 'garlics'], ['man', 'speaking'], ['little', 'boy', 'singing', 'playing', 'guitar'],

In [ ]:
print(text_st1)

[['girl', 'styling', 'hair'], ['group', 'men', 'play', 'soccer', 'beach'], ['one', 'woman', 'measuring', 'another', 'woman', 'ankle'], ['man', 'cutting', 'up', 'cucumber'], ['man', 'playing', 'harp'], ['woman', 'cutting', 'onions'], ['man', 'riding', 'electric', 'bicycle'], ['man', 'playing', 'drums'], ['man', 'playing', 'guitar'], ['man', 'playing', 'guitar'], ['man', 'playing', 'guitar'], ['man', 'cutting', 'onion'], ['man', 'cycling'], ['man', 'slicing', 'open', 'fish'], ['man', 'slicing', 'tomato'], ['man', 'playing', 'guitar'], ['baby', 'panda', 'goes', 'down', 'slide'], ['man', 'singing', 'playing', 'guitar'], ['man', 'attacks', 'woman'], ['man', 'driving', 'car'], ['woman', 'cutting', 'tofu'], ['woman', 'styling', 'hair'], ['two', 'zebras', 'play', 'open', 'field'], ['man', 'cutting', 'potato'], ['man', 'slicing', 'onion'], ['man', 'dancing'], ['man', 'riding', 'motorcycle'], ['woman', 'slicing', 'garlics'], ['man', 'speaking'], ['little', 'boy', 'singing', 'playing', 'guitar'],

In [ ]:
# ライブラリのインストール
! pip install gensim

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [ ]:
# 単語ベクトルの読み込み
import gensim.downloader as api
import gensim

en_w2v = api.load("word2vec-google-news-300")

#en_w2v = gensim.models.KeyedVectors.load_word2vec_format("/content/drive/MyDrive/exp5-3model/crawl-300d-2M.vec", binary=False)
#https://github.com/RaRe-Technologies/gensim-data

In [ ]:
#ベクトル中心化　*頻度
import numpy as np

vocab = en_w2v.wv.index2word
meanvec = np.zeros(en_w2v.vector_size)
for voc in set(prob_dict.keys()) & set(vocab):
    meanvec +=  en_w2v[voc] * prob_dict[voc]
print(meanvec[:10])

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:4: DeprecationWarning: Call to deprecated `wv` (Attribute will be removed in 4.0.0, use self instead).
  after removing the cwd from sys.path.


[ 0.00681015 -0.00865534 -0.04791635 -0.02049954  0.00717445 -0.01146622
 -0.02317206 -0.01860099 -0.02043321 -0.00533921]


In [ ]:
# 単語リストをベクトル化するための関数を定義

def vectorize(words, w2v):
    vectors = list()
    for word in words:
        if not word.isalpha():
            word = ''
        if word in w2v:
            vectors.append((w2v[word]-meanvec)*idf(word))
    return np.array(vectors).mean(axis=0)

In [ ]:
# 類似度推定: 2つのベクトル -> コサイン類似度

def cos(v1, v2):
    return np.dot(v1, v2) / (np.linalg.norm(v1) * np.linalg.norm(v2))

In [ ]:
y_nltk = list()
y_bow = list()
y_mt = list()
y_st = list()

for i in range(len(text_nltk1)):
    # nltk
    vector_nltk1 = vectorize(text_nltk1[i], en_w2v)
    vector_nltk2 = vectorize(text_nltk2[i], en_w2v)
    y_nltk.append('{:f}'.format(cos(vector_nltk1, vector_nltk2)))

    # bow
    vector_bow1 = vectorize_bow(text_nltk1[i], vocab, word2id)
    vector_bow2 = vectorize_bow(text_nltk2[i], vocab, word2id)
    y_bow.append('{:f}'.format(cos(vector_bow1, vector_bow2)))

    # Moses Tokenizer
    vector_mt1 = vectorize(text_mt1[i], en_w2v)
    vector_mt2 = vectorize(text_mt2[i], en_w2v)
    y_mt.append('{:f}'.format(cos(vector_mt1, vector_mt2)))

    # Stanza
    vector_st1 = vectorize(text_st1[i], en_w2v)
    vector_st2 = vectorize(text_st2[i], en_w2v)
    y_st.append('{:f}'.format(cos(vector_st1, vector_st2)))

In [ ]:
print(y_nltk)
print(y_bow)
print(y_mt)
print(y_st)
print(y_nltk == y_mt)
print(y_mt == y_st)

['0.790365', '0.952832', '0.994613', '0.848892', '0.560098', '0.542333', '0.915283', '0.683901', '0.859759', '0.644116', '0.644116', '1.000000', '0.306777', '0.706709', '0.684080', '0.695964', '0.894920', '0.874910', '0.636035', '0.327671', '0.524613', '0.384031', '0.963582', '0.808633', '0.663413', '0.909488', '0.722898', '0.694831', '0.431470', '0.924146', '0.806604', '0.859731', '0.365362', '0.860807', '0.734750', '0.266117', '0.385142', '0.651316', '0.942620', '0.393938', '0.853248', '0.883174', '0.874910', '0.375725', '0.552483', '0.226537', '0.190615', '0.477973', '0.860785', '0.681878', '0.518971', '0.575380', '0.360556', '0.182572', '0.813555', '0.339098', '0.939330', '0.350484', '0.574818', '0.655778', '1.000000', '0.139026', '0.796803', '0.358899', '0.768528', '0.371997', '0.323103', '0.751190', '0.182236', '0.636274', '0.846030', '0.965702', '0.267897', '0.811002', '0.712707', '0.299181', '0.343656', '0.124899', '0.707853', '0.866157', '0.210397', '0.456602', '0.454799', '0.

In [ ]:
# アンサンブル
y = list()
for i in range(len(y_nltk)):
    avg = float(y_nltk[i])*0 + float(y_bow[i])*3 + float(y_mt[i])*2 + float(y_st[i])*5
    y.append('{:f}'.format(avg))
print(y)

# ファイルへの書き込み
with open('/content/drive/MyDrive/exp5-3_english/eng.test.label.txt', 'w') as f:
    for i, label in enumerate(y):
        if i == len(y)-1:
            f.write(label)
        else:
            f.write(label+'\n')

['6.808077', '8.924693', '8.962589', '7.943808', '6.121853', '6.135349', '9.127955', '7.378122', '8.108341', '6.690197', '6.690197', '9.445880', '3.930120', '6.939665', '6.913679', '7.055009', '8.314048', '8.995501', '6.172638', '4.196360', '6.011736', '3.279164', '9.264835', '7.765959', '5.879391', '8.915153', '6.458697', '7.156887', '3.889709', '8.474783', '6.709591', '7.431484', '3.298834', '8.295284', '6.863127', '2.924831', '3.535230', '6.609377', '8.435246', '3.498859', '7.102023', '8.478866', '8.995501', '3.691114', '5.790314', '2.504845', '2.204916', '4.662475', '7.702789', '6.515633', '4.798839', '5.484149', '3.282797', '2.596979', '7.367041', '2.526055', '8.675543', '3.732155', '8.138392', '5.686385', '10.000000', '0.944678', '6.878326', '3.183836', '7.212987', '3.628674', '2.343593', '7.589564', '1.927877', '6.480736', '7.224453', '8.396987', '1.189804', '7.970343', '6.110867', '2.411388', '3.269825', '1.214745', '6.202719', '7.650156', '1.283583', '4.127399', '3.775700', '3

#スコアメモ

ベクトル中心化+idf+textfast(crawl-300d-2M.vec")+nltk正規化

* 0 : 3 : 2 : 6 : 0.744
* 0 : 3 : 2 : 5 : 0.746
* 0 : 2 : 2 : 5 : 0.739
* 1 : 3 : 2 : 5 : 0.744
* 0.1 : 3 : 2 : 5 : 0.745
* 5 : 3 : 2 : 5 : 0.738

ベクトル中心化+idf+fasttext

1:1:1:1 ： 0.740

0:3:2:5 ： 0.743

ベクトル中心化
*   0.00: 0.40: 0.20: 0.40  :  0.725

nltk : bow : mt : st
*   0.00: 0.20: 0.40: 0.40  :  0.730
*   0.00: 0.30: 0.10: 0.60  :  0.733
*   0.00: 0.30: 0.20: 0.50  :  0.734
*   0.00: 0.30: 0.30: 0.40  :  0.734
*   0.00: 0.30: 0.40: 0.30  :  0.734
*   0.00: 0.40: 0.50: 0.10  :  0.733
*   0.00: 0.40: 0.40: 0.20  :  0.734
*   0.00: 0.40: 0.30: 0.30  :  0.734
*   0.00: 0.40: 0.20: 0.40  :  0.734
*   0.00: 0.40: 0.10: 0.50  :  0.733
*   0.00: 0.50: 0.20: 0.30  :  0.730

nltk : bow : mt
*   0.00: 0.25: 0.75  :  0.731
*   0.00: 0.36: 0.64  :  0.733
*   0.00: 0.40: 0.60  :  0.733
*   0.00: 0.50: 0.50  :  0.729
*   0.36: 0.00: 0.64  :  0.705
*   0.02: 0.36: 0.62  :  0.733
*   0.12: 0.36: 0.52  :  0.733
*   0.22: 0.36: 0.42  :  0.732
*   0.32: 0.36: 0.32  :  0.731
*   0.42: 0.36: 0.22  :  0.731
*   0.52: 0.36: 0.12  :  0.730

nltk : bow
*   0.5: 0.5  :  0.725
*   0.6: 0.4  :  0.725
*   0.64: 0.36  : 0.728
*   0.65: 0.35  :  0.728
*   0.7: 0.3  :  0.727
*   0.75: 0.25  :  0.726
*   0.8: 0.2  :  0.723
*   0.9: 0.1  :  0.714




ストップワード[:26]
*   max:0.623
*   mean:0.647
*   min:0.656

ストップワード[:56]
*   max:0.591
*   mean:0.681
*   min:0.607

ストップワード[:65]
*   max:0.546
*   mean:0.701
*   min:0.561

ストップワード[:75]
*   max:0.524
*   mean:0.700
*   min:0.537

ストップワード[:85]
*   max:0.529
*   mean:0.701
*   min:0.541

ストップワード[:85]+'on'+'in'+"'s"
*   mean:0.703

ストップワード[:85]+'on'+'in'+"'s"+'one'+'two'
*   mean:0.702

y1 : y2  

y1 ：word2vec-google-news-300

y2：conceptnet-numberbatch-17-06-300

*   1.00 : 0.10 = 0.610
*   0.50 : 0.10 = 0.604
*   0.50 : 0.30 = 0.576
*   0.50 : 0.50 = 0.549
*   0.50 : 0.70 = 0.524
*   0.10 : 1.00 = 0.352


y1 ：word2vec-google-news-300

y2：glove-twitter-200

*   0.50 : 0.10 = 0.614
*   0.80 : 0.10 = 0.611
*   0.50 : 0.50 = 0.575
*   0.70 : 0.50 = 0.586
*   0.50 : 0.70 = 0.558





#延長戦

* word2vec-google-news-300 :  0.728
* glove-wiki-gigaword-300 :  0.716
* fasttext-wiki-news-subwords-300 :  0.705
* glove-twitter-200 :  0.712
* word2vec-ruscorpora-300 :  0.000 nan
* ParaNMTコーパスで学習された単語ベクトル : 0.775


